examples/energy-management-app/esp32/main/main.cpp

```diff
@@ -154,8 +154,7 @@ CHIP_ERROR DeviceEnergyManagementInit()
         BitMask<DeviceEnergyManagement::Feature, uint32_t>(
             DeviceEnergyManagement::Feature::kPowerAdjustment, DeviceEnergyManagement::Feature::kPowerForecastReporting,
             DeviceEnergyManagement::Feature::kStateForecastReporting, DeviceEnergyManagement::Feature::kStartTimeAdjustment,
-            DeviceEnergyManagement::Feature::kPausable, DeviceEnergyManagement::Feature::kForecastAdjustment,
-            DeviceEnergyManagement::Feature::kConstraintBasedAdjustment));
+            DeviceEnergyManagement::Feature::kPausable));
```

Comment:
not a blocker, but this couples the endpoint configuration and the example app - ie, if you return here that all the features are supported, then the zap conformance needs to match that. I mean, given that this IS an example app, that seems reasonable, but it might be worthwhile to note that in the readme somewhere (if it's not there already) because there is a tendency for folks to blindly copy the examples. Which, I suppose, is a problem in an of itself. Maybe a cert failure is what you get for blindly copying examples.

Reply:
I can add a comment here - but these should be greyed out in ZAP as the are implemented in the Read AttributeInterface handler https://github.com/project-chip/connectedhomeip/blob/e5f5d6798ed47706078462f1697c481a8dc60bfa/src/app/clusters/device-energy-management-server/device-energy-management-server.cpp#L97  and not set via ZAP.

This https://github.com/project-chip/connectedhomeip/blob/e5f5d6798ed47706078462f1697c481a8dc60bfa/src/app/zap-templates/zcl/zcl-with-test-extensions.json#L192 should be stopping ZAP from being editable.